cmd/av/pr.go

```diff
@@ -1,53 +1,97 @@
+	"github.com/aviator-co/av/internal/gh"
+	"github.com/aviator-co/av/internal/git"
+	"github.com/aviator-co/av/internal/meta"
+	"github.com/shurcooL/githubv4"
+	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
+	"io/ioutil"
+	"os"
+	"strings"
 )
 
 var prCmd = &cobra.Command{
 	Use: "pr",
 }
 
 var prCreateFlags struct {
-	Base string
+	Base   string
+	Force  bool
+	NoPush bool
+	Title  string
+	Body   string
 }
 var prCreateCmd = &cobra.Command{
 	Use:   "create",
 	Short: "create a pull request for the current branch",
+	Long: strings.TrimSpace(`
+Create a pull request for the current branch.
+
+Examples:
+  Create a PR with an empty body:
+    $ av pr create --title "My PR"
+
+  Create a pull request, specifying the body of the PR from standard input.
+    $ av pr create --title "Implement fancy feature" --body - <<EOF
+    > Implement my very fancy feature.
+    > Can you please review it?
+    > EOF
+`),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		// Lets keep this commented until we actually implement this :')
-		//if config.GitHub.Token == "" {
-		//	// TODO: lets include a documentation link here
-		//	logrus.Info(
-		//		"GitHub token is not configured. " +
-		//			"Consider adding it to your config file (at ~/.config/av/config.yaml) " +
-		//			"to allow av to automatically create pull requests.",
-		//	)
-		//}
-
-		repo, err := getRepo()
+		// arg validation
+		if prCreateFlags.Title == "" {
+			return errors.New("title is required")
+		}
+
+		repo, repoMeta, err := getRepoInfo()
 		if err != nil {
-			return errors.WrapIf(err, "failed to get repo")
+			return err
 		}
+
 		currentBranch, err := repo.CurrentBranchName()
 		if err != nil {
 			return errors.WrapIf(err, "failed to determine current branch")
 		}
-		origin, err := repo.Origin()
-		if err != nil {
-			return errors.WrapIf(err, "failed to determine origin")
+
+		if !prCreateFlags.NoPush {
+			// First, make sure we've set the upstream.
+			upstream, err := repo.RevParse(&git.RevParse{
+				SymbolicFullName: true,
+				Rev:              "HEAD@{u}",
+			})
+			if err != nil {
+				// Set the upstream branch
+				upstream = "refs/remotes/origin/" + currentBranch
+				if _, err := repo.Git("branch", "--set-upstream-to", upstream); err != nil {
+					return errors.WrapIf(err, "failed to set upstream")
+				}
+			}
+			logrus.WithField("upstream", upstream).Debug("pushing latest changes")
+			if _, err := repo.Git("push"); err != nil {
+				return errors.WrapIf(err, "failed to push")
+			}
+		}
+
+		// TODO:
+		//     It would be nice to be able to auto-detect that a PR has been
+		//     opened for a given PR without using av. We might need to do this
+		//     when creating PRs for a whole stack (e.g., when running `av pr`
+		//     on stack branch 3, we should make sure PRs exist for 1 and 2).
+		branchMeta, ok := meta.GetBranch(repo, currentBranch)
+		if ok && branchMeta.PullRequest.ID != "" && !prCreateFlags.Force {
```

Comment:
I'm assuming the GetBranch in future will populate `PullRequest.ID`?

Reply:
`meta.GetBranch` just loads the stored metadata. It already populates `PullRequest.ID` if it's been set (which is what we're checking here).